File: Data Structures & Algorithms/search-insert-position/submission-2.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        return bsearch(0, nums.size(), nums, target);
    }
    int bsearch(int l, int r, vector<int>& nums, int target){
        if(l>r){
            if(l>nums.size()){
                return l-1;
            }
            return l;
        }
        int m = l + (r-l) / 2;
        if(nums[m]==target){
            return m;
        }
        if(nums[m]>target){
            return bsearch(l,m-1, nums, target);
        }
        else{
            return bsearch(m+1, r, nums, target);
        }
        
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-2.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heap = []
        for num in nums:
            heapq.heappush(heap, -num)
            while len(heap)>len(nums)-k+1:
                heapq.heappop(heap)
        return -heap[0]

            